mozilla/blurts-server
# Mntor 1564 - Add Survey Recruitment Link

## Summary 

This PR adds a survey recruitment link to the top of the page for logged-in/English speaking users. 

## Acceptance Criteria 
- [x] Show to en-us only
- [x] Show only on the signed in dashboard page
- [x] Banner is dismissable

## References: 
Jira: [MNTOR-1564](https://mozilla-hub.atlassian.net/browse/MNTOR-1564)
Figma: N/A

## Testing

To show the recruitment link, three items must be true: 

- Set the following variables in `.env`:  
  - `RECRUITMENT_BANNER_LINK=https://survey.alchemer.com/s3/6785097/9ee8eccacd7b`
  - `RECRUITMENT_BANNER_TEXT=Tell us about your experience with Firefox Monitor and receive a $100 Amazon gift card.` 
- Set your primary browser language to an English variant (`en`, `en-us`, etc). 

To "reset" this banner, you'll need to delete a cookie. You can do that by: 
- Open Web Developer Tools on `http://localhost:6060` page
- Click on "Storage" tab
- Click "Cookies"
- Click `http://localhost:6060`
- Remove the `recruited` cookie

_Please reset your app between each test._ 

**Test: Click Survey Link**
- Run app locally, open homepage in logged out state 
- **Expected:** No banner should be visible
- Log in. Go to dashboard.
- **Expected:** Banner should be visible
- Click on text link "Tell us about…"
- **Expected:** New tab should open to survey link set in ENV
- Return to dashboard tab, refresh page
- **Expected:** Banner should no longer be visible

**Test: Click Dismiss Link on Survey**
- Run app locally, open homepage in logged out state 
- **Expected:** No banner should be visible
- Log in. Go to dashboard.
- **Expected:** Banner should be visible
- Click on X/dismiss button on the banner
- **Expected:** Banner should no longer be visible
- Refresh page
- **Expected:** Banner should still not be visible

**Test: Set Browser to non-English language**
- Run app locally, open homepage in logged out state 
- **Expected:** No banner should be visible
- Log in. Go to dashboard.
- **Expected:** Banner should be visible
- Set browser to German (`de`)
- Refresh page
- **Expected:** Banner should no longer be visible

## Checklist (Definition of Done)

## Screenshots

![image](https://github.com/mozilla/blurts-server/assets/2692333/3dacbe28-9870-4285-af3d-dec2c97f04d5)

![image](https://github.com/mozilla/blurts-server/assets/2692333/e69043f7-d478-4dcf-a3b6-31ec2cf05cba)
